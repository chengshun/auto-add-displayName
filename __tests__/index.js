var babel = require('babel-core');
var plugin = require('../main');
var classPlugin = require('babel-plugin-transform-class-properties');
var env = require('babel-preset-env');
var react = require('babel-preset-react');

var test1 = `
	class App extends Component {

	}
`;

var test2 = `
	class App extends Component {

	}

	class Test extends Component {

	}
`;

var test3 = `
	class App extends Component {
		static displayName = 'NoNeddAddDisplayName';
	}
`;

it('对单个class添加diaplayName', () => {
	const { code } = babel.transform(test1, {
		plugins: [classPlugin, plugin],
		presets: [react, env] 
	});

	expect(code).toMatchSnapshot();
});

it('对多个class添加diaplayName', () => {
	const { code } = babel.transform(test2, {
		plugins: [classPlugin, plugin],
		presets: [react, env] 
	});

	expect(code).toMatchSnapshot();
});

it('已有diaplayName，不需要添加', () => {
	const { code } = babel.transform(test3, {
		plugins: [classPlugin, plugin],
		presets: [react, env] 
	});

	expect(code).toMatchSnapshot();
});
