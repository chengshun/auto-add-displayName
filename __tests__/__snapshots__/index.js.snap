// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`对单个class添加diaplayName 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var App = function (_Component) {
	_inherits(App, _Component);

	function App() {
		_classCallCheck(this, App);

		return _possibleConstructorReturn(this, (App.__proto__ || Object.getPrototypeOf(App)).apply(this, arguments));
	}

	return App;
}(Component);

App.displayName = \\"App\\";"
`;

exports[`对多个class添加diaplayName 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var App = function (_Component) {
	_inherits(App, _Component);

	function App() {
		_classCallCheck(this, App);

		return _possibleConstructorReturn(this, (App.__proto__ || Object.getPrototypeOf(App)).apply(this, arguments));
	}

	return App;
}(Component);

App.displayName = \\"App\\";

var Test = function (_Component2) {
	_inherits(Test, _Component2);

	function Test() {
		_classCallCheck(this, Test);

		return _possibleConstructorReturn(this, (Test.__proto__ || Object.getPrototypeOf(Test)).apply(this, arguments));
	}

	return Test;
}(Component);

Test.displayName = \\"Test\\";"
`;

exports[`已有diaplayName，不需要添加 1`] = `
"'use strict';

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return call && (typeof call === \\"object\\" || typeof call === \\"function\\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function, not \\" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var App = function (_Component) {
	_inherits(App, _Component);

	function App() {
		_classCallCheck(this, App);

		return _possibleConstructorReturn(this, (App.__proto__ || Object.getPrototypeOf(App)).apply(this, arguments));
	}

	return App;
}(Component);

App.displayName = 'NoNeddAddDisplayName';"
`;
